packages/dd-trace/src/appsec/reporter.js

```diff
@@ -163,12 +184,14 @@ function getCollectedHeaders (req, res, shouldCollectEventHeaders, storedRespons
     )
   }
 
+  const maxHeadersCollected = extendedDataCollection?.max_collected_headers || config.maxHeadersCollected
```

Comment:
Nit: it would be useful to have a TODO comment here to remove `config.maxHeadersCollected` in the next major, as there are in the spots where deprectaed config is used.

Reply:
I prefer to have multiple todos, so we don't miss any of then in the future.